smalld/smalld.py

```diff
@@ -42,6 +42,16 @@ def all():
         return reduce(operator.ior, Intent.__members__.values())
 
 
+recoverable_error_codes = {
```

Comment:
I think I'd rather list unrecoverable error codes. i.e., if we encounter an error we don't know about, assume we can try reconnecting.

Reply:
With the limits set by discord I felt its unsafe to assume its recoverable especially that retrying a lot can get the token revoked or things like that (and IIRC some even result in a ban but correct me if I am wrong I am not sure of this.)

Reply:
It is true we don't want someone to get their token banned because of the lib. OTOH we should have rate limiting around those things that need rate limiting (e.g., identifying, sending web socket messages, etc). I guess until we have those it's fair to leave this.